gems/sorbet-runtime/lib/types/types/type_parameter.rb

```diff
@@ -3,11 +3,30 @@
 
 module T::Types
   class TypeParameter < Base
+    module Private
+      @pool = {}
+
+      def self.cached_entry(name)
+        @pool[name]
+      end
+
+      def self.set_entry_for(name, type)
+        @pool[name] = type
+      end
+    end
+
     def initialize(name)
       raise ArgumentError.new("not a symbol: #{name}") unless name.is_a?(Symbol)
       @name = name
     end
 
+    def self.make(name)
+      cached = Private.cached_entry(name)
+      return cached if cached
+
+      Private.set_entry_for(name, new(name))
+    end
+
```

Comment:
Should we add `private_class_method :new`?

Reply:
I thought about this, but decided since this class is publicly visible -- though I didn't find any bare usages of it in Stripe's codebase -- that I would leave this API alone.